rtos/Semaphore.cpp

```diff
@@ -27,7 +27,7 @@
 namespace rtos {
 
 Semaphore::Semaphore(int32_t count) {
-    constructor(count, 1024);
+    constructor(count, UINT16_MAX);
```

Comment:
Actually including stdint.h didn't work. UINT16_MAX isn't defined until c++11/c99. I've just updated to a hex literal.